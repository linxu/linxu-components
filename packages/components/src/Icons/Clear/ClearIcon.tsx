import React, { MouseEventHandler } from "react";
import { ClearIconSvg } from "./ClearIconSvg";
import {Colors} from "../../theme/Colors";

export interface ClearIconProps {
  width?: string | number;
  height?: string | number;
  transparent?: boolean;
  style?: React.CSSProperties;
  color?: keyof Colors;
  onClick?: MouseEventHandler | undefined;
  "data-testid"?: string;
}

export const ClearIcon = ({
  transparent,
  width,
  height,
  style,
  color,
  onClick,
  ...rest
}: ClearIconProps & any) => (
  <ClearIconSvg
    width={width || "24"}
    height={height || "24"}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
    style={style}
    color={color}
    onClick={onClick}
    data-testid={rest["data-testid"]}
    {...rest}
  >
    {!transparent && <circle cx="10" cy="10" r="10" fill="white" />}
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12ZM7.75732 9.17163L9.17154 7.75742L12 10.5858L14.8284 7.75742L16.2426 9.17163L13.4142 12.0001L16.2426 14.8285L14.8284 16.2427L12 13.4143L9.17154 16.2427L7.75732 14.8285L10.5858 12.0001L7.75732 9.17163Z"
    />
  </ClearIconSvg>
);
