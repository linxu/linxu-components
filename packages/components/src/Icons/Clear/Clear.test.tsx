import React from "react";
import { render } from "@testing-library/react";

import { ClearIcon } from "./ClearIcon";

it("renders", async () => {
  const tree = render(<ClearIcon />);
  expect(tree).toMatchSnapshot();
});
