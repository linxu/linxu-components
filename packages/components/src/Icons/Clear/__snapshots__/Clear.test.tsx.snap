// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="sc-bdfBQB kleLNI"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="10"
          cy="10"
          fill="white"
          r="10"
        />
        <path
          clip-rule="evenodd"
          d="M23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12ZM7.75732 9.17163L9.17154 7.75742L12 10.5858L14.8284 7.75742L16.2426 9.17163L13.4142 12.0001L16.2426 14.8285L14.8284 16.2427L12 13.4143L9.17154 16.2427L7.75732 14.8285L10.5858 12.0001L7.75732 9.17163Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="sc-bdfBQB kleLNI"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="10"
        cy="10"
        fill="white"
        r="10"
      />
      <path
        clip-rule="evenodd"
        d="M23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12ZM7.75732 9.17163L9.17154 7.75742L12 10.5858L14.8284 7.75742L16.2426 9.17163L13.4142 12.0001L16.2426 14.8285L14.8284 16.2427L12 13.4143L9.17154 16.2427L7.75732 14.8285L10.5858 12.0001L7.75732 9.17163Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
